### PR TITLE
feat(stages): add table checkpoint to `AccountHashing` and `StorageHashing`

### PR DESCRIPTION
**Pull request by joshieDo**
Similar to/blocked by/branch out of #1656

Removes `tx.commit` from the hashing stages and uses checkpoints through `tables::SyncStageProgress`

### CODE REVIEW

File: crates/stages/src/stages/hashing_account.rs [hunk 1]
```diff
@@ -137,43 +175,57 @@ impl<DB: Database> Stage<DB> for AccountHashingStage {
...
+            let mut checkpoint = self.get_checkpoint(tx)?;
+
+            if checkpoint.address.is_none() ||
+                checkpoint.to != to_transition ||
+                checkpoint.from != from_transition
+            {
+                // clear table, load all accounts and hash it
+                tx.clear::<tables::HashedAccount>()?;
+
+                // Checkpoint is invalid if the range changed
```
Comment (gakonst) [inline crates/stages/src/stages/hashing_account.rs:187]: what does this comment mean? seems unrelated?

File: crates/stages/src/stages/hashing_storage.rs
Comment (gakonst) [inline crates/stages/src/stages/hashing_storage.rs:1]: beautiful, really

File: crates/stages/src/stages/merkle.rs
Comment (gakonst) [inline crates/stages/src/stages/merkle.rs:1]: lets separate the 2 PRs from each other

File: crates/stages/src/stages/hashing_account.rs (hunk 1 above)
Comment (joshieDo) [reply crates/stages/src/stages/hashing_account.rs]: tried adding a better explanation and move it to a more relevant line. does it make more sense  ?

File: crates/stages/src/stages/hashing_storage.rs [hunk 2]
```diff
@@ -57,77 +95,92 @@ impl<DB: Database> Stage<DB> for StorageHashingStage {
...
+                                    // is None
+                                    let keccak_address =
+                                        if let Some(keccak_address) = keccak_address {
+                                            keccak_address
+                                        } else {
+                                            keccak256(address)
+                                        };
+
+                                    // TODO cache map keccak256(slot.key) ?
+                                    ((keccak_address, keccak256(slot.key)), slot.value)
```
Comment (gakonst) [inline crates/stages/src/stages/hashing_storage.rs:137]: yeah probably right solution here is just to have a LRU maps for map<address, keccak256(address)> and map<h256, keccak256(h256)> to avoid redundant hashing, but doubt it'll be a bottleneck

**Review by gakonst — APPROVED** [review]
